test/unit/dataMap/metaManager/utils.spec.js

```diff
@@ -0,0 +1,177 @@
+import { expandMetaType, columnFactory, isUnsignedNumber, assert, isNullish } from 'handsontable/dataMap/metaManager/utils';
+
+describe('MetaManager utils', () => {
+  describe('expandMetaType', () => {
+    it('should return "undefined" when an object doesn\'t have defined "type" property or is not supported', () => {
+      expect(expandMetaType()).toBeUndefined();
+      expect(expandMetaType(null)).toBeUndefined();
+      expect(expandMetaType(1)).toBeUndefined();
+      expect(expandMetaType(true)).toBeUndefined();
+      expect(expandMetaType(NaN)).toBeUndefined();
+      expect(expandMetaType([])).toBeUndefined();
+    });
+
+    it('should return only properties that are not defined on the child object', () => {
+      class Child {
+        constructor() {
+          this.copyPaste = false;
+        }
+      }
+
+      expect(expandMetaType({ copyPaste: true, test: 'foo' }, new Child())).toEqual({ test: 'foo' });
+
+      expect(expandMetaType('autocomplete', { editor: () => {} })).toEqual({
+        renderer: expect.any(Function),
+        validator: expect.any(Function),
+      });
+    });
+
+    it('should return a copy of the object that is holding by "type" property', () => {
+      const type = { copyPaste: true, test: 'foo' };
+
+      expect(expandMetaType(type)).not.toBe(type);
+      expect(expandMetaType(type)).toEqual(type);
+    });
+
+    it('should return the object with defined properties defined by "autocomplete" cell type', () => {
+      expect(expandMetaType('autocomplete')).toEqual({
+        editor: expect.any(Function),
+        renderer: expect.any(Function),
+        validator: expect.any(Function),
+      });
+    });
+
+    it('should return the object with defined properties defined by "checkbox" cell type', () => {
+      expect(expandMetaType('checkbox')).toEqual({
+        editor: expect.any(Function),
+        renderer: expect.any(Function),
+      });
+    });
+
+    it('should return the object with defined properties defined by "date" cell type', () => {
+      expect(expandMetaType('date')).toEqual({
+        editor: expect.any(Function),
+        renderer: expect.any(Function),
+        validator: expect.any(Function),
+      });
+    });
+
+    it('should return the object with defined properties defined by "dropdown" cell type', () => {
+      expect(expandMetaType('dropdown')).toEqual({
+        editor: expect.any(Function),
```

Comment:
How do we know that any of this `expandMetaType` calls has set the right functions? They may be the same for all cell types or mixed as well 🤔

Reply:
Here I checked generally the shape of the object. More specific tests are written here https://github.com/handsontable/handsontable/blob/master/test/e2e/cellTypes/index.spec.js.